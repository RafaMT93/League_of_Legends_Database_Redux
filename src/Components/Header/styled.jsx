import styled from 'styled-components';

export const WrapperHeader = styled.header`
  background-color: ${({ theme }) => theme.colors.header};
  height: 4rem;
  padding: 1rem;
  color: ${({ theme }) => theme.colors.primary};
`;

export const WrapperNav = styled.nav`
  display: flex;
  justify-content: space-between;
  align-itens: center;
  font-weight: bold;
`;
