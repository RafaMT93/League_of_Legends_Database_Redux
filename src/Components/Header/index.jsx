import React from 'react';
import { Link } from 'react-router-dom';
import { WrapperHeader, WrapperNav } from './styled';

const Header = () => {
  return (
    <WrapperHeader>
      <WrapperNav className="container">
        <Link to="/">Logo</Link>
        <Link to="/">Champions</Link>
        <Link to="/Item">Itens</Link>
      </WrapperNav>
    </WrapperHeader>
  );
};

export default Header;
