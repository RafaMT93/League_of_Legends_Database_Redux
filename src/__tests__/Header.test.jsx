import React from 'react';
import Header from '../Components/Header';
import '@testing-library/jest-dom';

import { BrowserRouter } from 'react-router-dom';

import { render, screen } from '@testing-library/react';
import theme from '../Global/Theme';
import { ThemeProvider } from 'styled-components';

const options = ['Champions', 'Items', 'ChampionList'];

describe('Component Header', () => {
  it('Render correctly', () => {
    render(
      <ThemeProvider theme={theme}>
        <BrowserRouter>
          <Header options={options} />
        </BrowserRouter>
      </ThemeProvider>,
    );

    expect(screen.queryByText(options[0])).toMatchSnapshot();
    expect(screen.queryByText(options[1])).toMatchSnapshot();
    expect.not.objectContaining(screen.queryByText(options[2]));
  });
});
