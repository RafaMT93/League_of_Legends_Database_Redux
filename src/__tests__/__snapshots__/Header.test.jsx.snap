// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Header Render correctly 1`] = `
<a
  href="/"
>
  Champions
</a>
`;

exports[`Component Header Render correctly 2`] = `null`;
